src/sentry/utils/versioning.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import absolute_import
+
+import warnings
+from collections import namedtuple
+
+from sentry.exceptions import InvalidConfiguration
+
+
+class Version(namedtuple('Version', 'major minor patch')):
+    def __str__(self):
+        return '.'.join(map(str, self))
+
+
+def make_upgrade_message(service, modality, version, hosts):
+    return '{service} {modality} be upgraded to {version} on {hosts}.'.format(
+        hosts=', '.join('{0} (currently {1})'.format(*i) for i in hosts),
```

Comment:
> imo, it's probably not valuable to know what each version is running

Fair enough. I thought it might be a useful to know how far behind a version was (especially for deprecations, rather than hard requirements.) This message is probably the wrong place to do that — it should probably be pushed into the web admin later.